src/loop_in_chunks.cpp

```diff
@@ -358,7 +358,18 @@ void fields::loop_in_chunks(field_chunkloop chunkloop, void *chunkloop_data, con
   vec s0(gv.dim), e0(gv.dim), s1(gv.dim), e1(gv.dim);
   compute_boundary_weights(gv, where, is, ie, snap_empty_dimensions, s0, e0, s1, e1);
 
+  int original_vol = 1;
+  LOOP_OVER_DIRECTIONS(gv.dim, d){
+    if (boundaries[High][d] == Periodic && boundaries[Low][d] == Periodic) {
+      original_vol *= (ie.in_direction(d) - is.in_direction(d))/2+1;
+    } else {
+      int vol_size_d = (min(user_volume.big_owned_corner(cgrid), ie).in_direction(d) - max(user_volume.little_owned_corner(cgrid),is).in_direction(d))/2+1;
```

Comment:
Sometimes, ``where`` doesn't intersect with the ``user_volume``, so the size on that direction is negative.